catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeValueSerializer.java

```diff
@@ -24,8 +24,8 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link org.codice.ddf.catalog.ui.forms.model.FilterNodeImpl} requires polymorphic behavior which
- * causes issues with JSON serialization. To avoid values of the following complex form:
+ * {@link IntermediateFilterNode} requires polymorphic behavior which causes issues with JSON
+ * serialization. To avoid values of the following complex form:
```

Comment:
❓ Do we need to update the `FilterNodeValueSerializer` to check the `property` and `value` fields for `instanceof Map` or otherwise complex data structures and defer processing if that's the case?

Reply:
I haven't need to in order to get the UI to function as expected. But let's talk about it.